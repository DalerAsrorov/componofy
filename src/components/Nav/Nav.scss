.number-badge {
  span {
    margin-top: 4px;
    margin-right: -6px;
  }
}
