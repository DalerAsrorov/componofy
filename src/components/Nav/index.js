import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Tabs, { Tab } from 'material-ui/Tabs';
import PhoneIcon from 'material-ui-icons/Phone';
import FavoriteIcon from 'material-ui-icons/Favorite';
import PersonPinIcon from 'material-ui-icons/PersonPin';
import HelpIcon from 'material-ui-icons/Help';
import ShoppingBasket from 'material-ui-icons/ShoppingBasket';
import ThumbDown from 'material-ui-icons/ThumbDown';
import ThumbUp from 'material-ui-icons/ThumbUp';

const styles = theme => ({
    root: {
        width: '100%',
        backgroundColor: theme.palette.background.paper
    }
});

class Nav extends PureComponent {
    static propTypes = {
        classes: PropTypes.object
    };

    _handleChange = (event, value) => {
        event.preventDefault();
        console.log(event, value);
    };

    render() {
        const { classes } = this.props;

        return (
            <div className={classes.root}>
                <AppBar position="static">
                    <Tabs
                        value={1}
                        onChange={this._handleChange}
                        scrollable
                        scrollButtons="off"
                    >
                        <Tab icon={<PhoneIcon />} />
                        <Tab icon={<FavoriteIcon />} />
                        <Tab icon={<PersonPinIcon />} />
                        <Tab icon={<HelpIcon />} />
                        <Tab icon={<ShoppingBasket />} />
                        <Tab icon={<ThumbDown />} />
                        <Tab icon={<ThumbUp />} />
                    </Tabs>
                </AppBar>
            </div>
        );
    }
}

export default withStyles(styles)(Nav);
