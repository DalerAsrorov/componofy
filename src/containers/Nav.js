import { withRouter } from 'react-router-dom';
import { connectStream } from '../connectPage';
import Nav from '../components/Nav';

export default connectStream(withRouter(Nav));
