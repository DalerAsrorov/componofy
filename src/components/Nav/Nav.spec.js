import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Nav from './';

Enzyme.configure({ adapter: new Adapter() });

// Classes required for Material UI theme
const classes = {
  root: '',
  tabContainer: '',
};

// Required props for Material UI components context
const requiredProps = {
  numberOfFinalPlaylists: 30,
  location: {
    pathname: 0,
  },
  navigation: {
    0: 'route1',
    1: 'route2',
    routeToIndexMap: jest.fn(),
    index: 0,
  },
  setNavIndex: jest.fn(),
  navigateTo: jest.fn(),
  classes,
};

describe('When Nav component is initialized', () => {
  let mount, shallow;

  beforeEach(() => {
    mount = createMount();
    shallow = createShallow();
  });

  it('Nav renders properly', () => {
    let wrapper = createShallow(<Nav {...requiredProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Nav has correct classes for styling', () => {
    let wrapper = shallow(<Nav {...requiredProps} />);
    let classNames = wrapper.prop('classes');

    Object.keys(classes).forEach((stylAttr) => {
      expect(classNames).toHaveProperty(stylAttr);
    });
  });

  it('Nav has three tabs', () => {
    let wrapper = mount(<Nav {...requiredProps} />);

    expect(wrapper.find(Tabs)).toHaveLength(1);
    expect(wrapper.find(Tab)).toHaveLength(3);
  });

  it('should show badge with the number of playlsits added', () => {
    let expected = 10;
    let wrapper = mount(
      <Nav {...requiredProps} numberOfFinalPlaylists={expected} />
    );
    let badge = wrapper.find('.number-badge').at(0);

    expect(badge.props().badgeContent).toBe(expected);
  });

  it('should not show badge if number of playlists added is 0', () => {
    let expected = false;
    let wrapper = mount(<Nav {...requiredProps} numberOfFinalPlaylists={0} />);
    let badge = wrapper.find('.number-badge');

    expect(badge.exists()).toEqual(expected);
  });

  it('should initially have the tab index value of 0', () => {
    let wrapper = mount(<Nav {...requiredProps} />);
    let tabsWrapper = wrapper.find(Tabs);
    let expected = 0;

    expect(tabsWrapper.props().value).toBe(0);
  });
});
