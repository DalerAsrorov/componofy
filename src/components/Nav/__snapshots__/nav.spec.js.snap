// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Nav component is initialized Nav renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Nav)
    classes={
      Object {
        "root": "",
        "tabContainer": "",
      }
    }
    location={
      Object {
        "pathname": 0,
      }
    }
    navigateTo={[Function]}
    navigation={
      Object {
        "0": "route1",
        "1": "route2",
        "index": 0,
        "routeToIndexMap": [Function],
      }
    }
    numberOfFinalPlaylists={30}
    setNavIndex={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "root": "Nav-root-1",
        "tabContainer": "Nav-tabContainer-2",
      },
      "location": Object {
        "pathname": 0,
      },
      "navigateTo": [Function],
      "navigation": Object {
        "0": "route1",
        "1": "route2",
        "index": 0,
        "routeToIndexMap": [Function],
      },
      "numberOfFinalPlaylists": 30,
      "setNavIndex": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "root": "Nav-root-1",
          "tabContainer": "Nav-tabContainer-2",
        },
        "location": Object {
          "pathname": 0,
        },
        "navigateTo": [Function],
        "navigation": Object {
          "0": "route1",
          "1": "route2",
          "index": 0,
          "routeToIndexMap": [Function],
        },
        "numberOfFinalPlaylists": 30,
        "setNavIndex": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
